integration/src/test/java/org/broadleafcommerce/core/payment/service/NullIntegrationGatewayTransactionServiceImpl.java

```diff
@@ -158,9 +159,9 @@ protected PaymentResponseDTO commonCreditCardProcessing(PaymentRequestDTO reques
                 String expMonth = parsedDate[0];
                 String expYear = parsedDate[1];
                 try {
-                    DateTime expirationDate = new DateTime(Integer.parseInt("20" + expYear), Integer.parseInt(expMonth), 1, 0, 0);
-                    expirationDate = expirationDate.dayOfMonth().withMaximumValue();
-                    validDate = expirationDate.isAfterNow();
+                    Date expirationDate = new Date(Integer.parseInt("20" + expYear), Integer.parseInt(expMonth), 1, 0, 0);
+                    expirationDate.toInstant().atZone(ZoneId.systemDefault()).toLocalDateTime();
```

Comment:
are you sure that you don't need to assign result of expirationDate.toInstant().atZone(ZoneId.systemDefault()).toLocalDateTime();  ?